modules/basegl/src/processors/entryexitpointsprocessor.cpp

```diff
@@ -58,22 +58,40 @@ EntryExitPoints::EntryExitPoints()
     addProperty(camera_);
     addProperty(trackball_);
 
-    for (auto& shader : entryExitHelper_.getShaders()) {
-        shader.get().onReload([this]() { invalidate(InvalidationLevel::InvalidResources); });
-    }
+    callback_ =
+        entryExitHelper_.onReload([this]() { invalidate(InvalidationLevel::InvalidResources); });
 }
 
 EntryExitPoints::~EntryExitPoints() = default;
 
 void EntryExitPoints::process() {
-    if (!entryImg_ || !entryImg_->isValid()) {
-        entryImg_ = entryPort_.getEditableData()->getEditableRepresentation<ImageGL>();
+    auto entry = entryPort_.getEditableData();
+
+    if (inport_.getData()->hasBuffer(BufferType::NormalAttrib) &&
+        entry->getNumberOfColorLayers() != 2) {
+
+        entry = std::make_shared<Image>(entry->getDimensions(), DataVec4UInt16::get());
+        entry->addColorLayer(
```

Comment:
Add comment that this layer is for the normals. 8-bit integer should be enough when they are axis-aligned. Precisison is perhaps not too important since they are only used once and arbitrary clipping is not too common.

Reply:
Since the normals are just just for shading usually I don't think it should matter very much, And in the 4k stereo rendering, we might not want to waste full image textures...